controllers/helmchart_controller.go

```diff
@@ -467,13 +469,24 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 		opts.VersionMetadata = strconv.FormatInt(obj.Generation, 10)
 	}
 
+	keyring, err := r.getProvenanceKeyring(ctx, obj)
+	if err != nil {
+		e := &serror.Event{
+			Err:    fmt.Errorf("failed to get public key for chart signature verification: %w", err),
+			Reason: sourcev1.AuthenticationFailedReason,
```

Comment:
The doc of `AuthentiactionFailedReason`, states that :
```
AuthenticationFailedReason signals that a Secret does not have the required fields, or the provided credentials do not match.
```
This seems like the appropriate reason to here, because `getProvenanceKeyring` returns an error only when it can't find a secret or the secret doesn't have the required key.

Reply:
Yes, but only if the Secret is related to authentication, which provenance isn't.